instance/models/load_balancer.py

```diff
@@ -183,24 +193,28 @@ def run_playbook(self, ansible_vars):
         This is factored out into a separate method so it can be mocked out in the tests.
         """
         playbook_path = pathlib.Path(settings.SITE_ROOT) / "playbooks/load_balancer_conf/load_balancer_conf.yml"
-        returncode = ansible.capture_playbook_output(
-            requirements_path=str(playbook_path.parent / "requirements.txt"),
-            inventory_str=self.domain,
-            vars_str=ansible_vars,
-            playbook_path=str(playbook_path),
-            username=self.ssh_username,
-            logger_=self.logger,
-        )
+        with cache.lock(self.domain):
```

Comment:
@mtyaka Since we are using a context manager here, the process would have to die abruptly to prevent the clean-up from being run, but of course this can happen (e.g. power outage).  I've added a timeout of 15 minutes to the lock, so it will get released in any case.  (The reconfiguration typically takes less than one minute, and we might further improve this by caching the virtualenv.)

I also used the more specific name for the lock.